dask_expr/_repartition.py

```diff
@@ -131,6 +144,13 @@ class RepartitionToFewer(Repartition):
     def _divisions(self):
         return tuple(self.frame.divisions[i] for i in self._partitions_boundaries)
 
+    def _lower(self):
+        # Multiple lowering steps might have changed the npartitions
+        npartitions = self.new_partitions
+        npartitions_input = self.frame.npartitions
+        if npartitions_input <= npartitions:
+            return self.frame
```

Comment:
This seems more like a simplify call to me?  Why here?  Why aren't simplify calls being used at this stage?

Also, slight style nit, but I would probably avoid the temporary variables here and just write this as

```python
if self.new_partitions <= self.frame.npartitions:
    return self.frame
```

Also, I'm surprised that this isn't `==`.  Does this mean that we're not supposed to reduce the number of partitions?

Reply:
Yeah I don't like this pattern more than you do.

The check is the other way round,

```
if self.frame.npartitions <= self.new_partitions:
    return self.frame
```

Lower might reduce the number of partitions of self.frame, which might make something that has been a repartition to fewer call earlier into a no-op. We are lowering from the root, which makes us vulnerable to things like this (sort_values is an example that will reduce the number of partitions)

Reply:
Sorry, I don't understand yet.  The situation I think you're describing is when we're calling `Repartition(X, 10)` where X has, say, 20 partitions. But then suddenly X has 5 partitions and so we're saying "well we no longer need to repartition X to 10 partitions, it's already at the nice low value of 5".  Is that right?

If so, I can see how this would make sense if our goal is to reduce partition counts, but this doesn't make sense to me in general.  What if someone wants precisely 10 partitions because they have ten machines or whatever and they want to have one partition per machine, no matter what.